server/src/main/java/org/opensearch/action/ingest/SimulatePipelineTransportAction.java

```diff
@@ -88,6 +88,8 @@ protected void doExecute(Task task, SimulatePipelineRequest request, ActionListe
             return;
         }
 
+        ingestService.validateProcessorCountForIngestPipeline(simulateRequest.getPipeline());
```

Comment:
Can you add a test for the validation in simulate flow?

Reply:
Also, please see if SimulateExecutionService is a better fit for placing this check.

Reply:
Updated.